examples/rbm_image_generation.py

```diff
@@ -0,0 +1,214 @@
+
+    # Plot grid of images
+    fig, axs = plt.subplots(rows, columns, figsize=(columns, rows))
+
+    idx = 0
+    for r in range(rows):
+        for c in range(columns):
+            axs[r, c].imshow(samples[idx], cmap="gray")
+            axs[r, c].axis("off")
+            idx += 1
+
+    fig.suptitle("Generated images from RBM trained on MNIST", fontsize=18)
+    plt.tight_layout(rect=[0, 0, 1, 0.96])
+    plt.savefig(output_path, bbox_inches="tight", pad_inches=0.1)
+    plt.show()
+    print(f"Generated {num_images} samples in {output_dir}/{output_filename}")
+
+
+def train_rbm(
+    n_visible: int = 784,
+    n_hidden: int = 500,
+    n_gibbs_steps: int = 10,
+    learning_rate: float = 1e-3,
+    momentum: float = 0.5,
+    weight_decay: float = 1e-7,
+    n_epochs: int = 20,
+    batch_size: int = 32,
+    dataset_path: str = "data",
+    output_dir: str = "samples",
+    output_filename: str = "generated_images.png",
+) -> None:
+    """Train an RBM on MNIST and generate sample images.
+
+    Args:
+        n_visible (int, optional): Number of visible units. Defaults to 784.
+        n_hidden (int, optional): Number of hidden units. Defaults to 500.
+        n_gibbs_steps (int, optional): Number of Gibbs sampling steps per CD update. Defaults to 10.
+        learning_rate (float, optional): Base learning rate for CD updates. Defaults to 1e-3.
+        momentum (float, optional): Momentum coefficient for CD updates. Defaults to 0.5.
+        weight_decay (float, optional):  Weight decay (L2 regularization) coefficient. Defaults to 1e-7.
+        n_epochs (int, optional): Number of training epochs. Defaults to 20.
+        batch_size (int, optional): Batch size for training. Defaults to 32.
+        dataset_path (str, optional): Path to download/store the MNIST dataset. Defaults to "data".
+        output_dir (str, optional): Directory to save the generated images. Defaults to "samples".
+        output_filename (str, optional): File name for saving the generated image grid. Defaults to "generated_images.png".
+    """
+
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    print(f"Using {device} device")
+
+    # Load MNIST data
+    print("Loading MNIST dataset...")
+    train_dataset = load_binarized_mnist(dataset_path)
+
+    # Create data loader
+    train_loader = torch.utils.data.DataLoader(
+        train_dataset, batch_size=batch_size, shuffle=True, drop_last=True
+    )
+
+    # Initialize RBM
+    rbm = RBM(n_visible, n_hidden).to(device)
+
+    # Train RBM
+    train_loop(
+        train_loader,
+        rbm,
+        n_epochs=n_epochs,
+        n_gibbs_steps=n_gibbs_steps,
+        learning_rate=learning_rate,
+        momentum=momentum,
+        weight_decay=weight_decay,
+    )
+
+    # Generate and save samples
+    generate_and_save_images(
+        rbm,
+        rows=8,
+        columns=8,
+        steps=1000,
+        output_dir=output_dir,
+        output_filename=output_filename,
+    )
```

Comment:
Maybe move this to the main section below, after `train_rbm`, since it's basically stand alone.